analytics/tracker.go

```diff
@@ -230,19 +236,33 @@ func (t tracker) SendStepFinishedEvent(properties analytics.Properties, result S
 		return
 	}
 
-	var eventName string
-	var extraProperties analytics.Properties
+	var (
+		eventName       string
+		extraProperties analytics.Properties
+	)
 
 	switch result.Status {
 	case models.StepRunStatusCodeSuccess:
 		eventName = stepFinishedEventName
 		extraProperties = analytics.Properties{statusProperty: successfulValue}
-		break
 	case models.StepRunStatusCodeFailed, models.StepRunStatusCodeFailedSkippable:
 		eventName = stepFinishedEventName
 		extraProperties = analytics.Properties{statusProperty: failedValue}
 		extraProperties.AppendIfNotEmpty(errorMessageProperty, result.ErrorMessage)
-		break
+	case models.StepRunStatusAbortedTimeout:
+		eventName = stepAbortedEventName
+		extraProperties = analytics.Properties{reasonProperty: customTimeoutValue}
+
+		if result.Timeout >= 0 {
+			extraProperties[timeoutProperty] = result.Timeout.Seconds()
+		}
+	case models.StepRunStatusAbortedNoOutputTimeout:
+		eventName = stepAbortedEventName
+		extraProperties = analytics.Properties{reasonProperty: noOutputTimeoutValue}
+
+		if result.Timeout >= 0 {
+			extraProperties[timeoutProperty] = result.Timeout.Seconds()
```

Comment:
I find it a bit confusing, isn't it the step timeout? In this case the step was not abourted because of that step timeout but because of the no output timeout (which will also be user configurable, so we might want to log that)

Reply:
Separated into a different field.